upi/openstack/down-02_network.yaml

```diff
@@ -9,29 +10,43 @@
   gather_facts: no
 
   tasks:
+  - name: 'List ports attached to router'
+    command:
+      cmd: "openstack port list --device-owner=network:ha_router_replicated_interface --tags {{ cluster_id_tag }} -f value -c id"
+    register: router_ports
+
   - name: 'Remove the ports from router'
-    os_delete_ports_on_router:
-      router: "{{ os_router }}"
-      tags: "{{ [cluster_id_tag] }}"
-    when: os_networking_type == "Kuryr"
+    command:
+      cmd: "openstack router remove port {{ os_router }} {{ item.1}}"
+    with_indexed_items: "{{ router_ports.stdout_lines }}"
+
+  - name: 'List ports'
+    command:
+      cmd: "openstack port list --tags {{ cluster_id_tag }} -f value -c id "
+    register: ports
 
   - name: 'Remove the cluster ports'
-    os_delete_network_resources:
-      resource: "port"
-      tags: "{{ [cluster_id_tag] }}"
+    command:
+      cmd: "openstack port delete {{ item.1}}"
+    with_indexed_items: "{{ ports.stdout_lines }}"
 
   - name: 'Remove the cluster router'
     os_router:
       name: "{{ os_router }}"
       state: absent
 
+  - name: 'List cluster networks'
+    command:
+      cmd: "openstack network list --tags {{ cluster_id_tag }} -f value -c Name"
```

Comment:
nit: as multiple networks can have the same name, you might want to `-c ID` instead to avoid issues

Reply:
oh no, forget that, Ansible will never want to deal with unpredictable IDs :(